src/foraging_gui/MyFunctions.py

```diff
@@ -890,7 +897,56 @@ def _LickSta(self,Trials=None):
             self.Start_CoCue_LeftRightRatio=np.nan
         else:
             self.Start_CoCue_LeftRightRatio=np.array(sum(self.Start_GoCue_LeftLicks))/np.array(sum(self.Start_GoCue_RightLicks))
-        
+
+    def calculate_inter_lick_intervals(self):
+        """
+        Calculate and categorize lick intervals
+        """
+
+        right = self.B_RightLickTime
+        left = self.B_LeftLickTime
+        threshold = .1
+
+        if right is not None and left is not None and len(right) > 0 and len(left) > 0:
+            # calculate left interval and fraction
+            same_side_l = np.diff(left)
+            same_side_l_frac = len(same_side_l[~(same_side_l > threshold)]) / len(same_side_l)
+            logging.info(f'Percentage of left lick intervals under 100 ms is {same_side_l_frac * 100}%.')
+            self.B_LeftLickIntervalPercent = same_side_l_frac * 100
+
+            # calculate right interval and fraction
+            same_side_r = np.diff(right)
+            same_side_r_frac = len(same_side_r[~(same_side_r > threshold)]) / len(same_side_r)
+            logging.info(f'Percentage of right lick intervals under 100 ms is {same_side_r_frac * 100}%.')
+            self.B_RightLickIntervalPercent = same_side_r_frac * 100
+
+            if same_side_l_frac + same_side_r_frac >= .1:
+                self.win.same_side_lick_interval.setText(f'Percentage of same side lick intervals under 100 ms is '
+                                                         f'over 10%: {(same_side_r_frac+same_side_l_frac) * 100}%.')
+            else:
+                self.win.same_side_lick_interval.setText('')
+
+            # calculate cross side interval and frac
+            dummy_array = np.ones(right.shape)  # use array to tell lick side after merging
+            stacked_right = np.column_stack((dummy_array, right))
+            stacked_left = np.column_stack((np.negative(dummy_array), left))
+            merged_sorted = sorted(np.concatenate((stacked_right, stacked_left)),
+                                   key=lambda x: x[1])  # sort by second element
+            cross_sides = []
+            for i in range(1, len(right)):
+                if merged_sorted[i - 1][0] == -merged_sorted[i][0]:  # cross side lick
+                    cross_sides.append(merged_sorted[i][1] - merged_sorted[i - 1][1])
+            cross_sides = np.array(cross_sides)
+            cross_side_frac = len(cross_sides[~(cross_sides > threshold)]) / len(cross_sides)
```

Comment:
This code block is very confusing, and I'm not totally sure whats going on. 

I think I would have implemented this with a while loop and a pointer in each of the left and right lick times

Reply:
Ok, I understand whats happening now and it makes sense. I would add a few comments so we can understand how this works in the future. Just explain the structure of `merged_sorted` , and `cross_sides` (which contains only the ITIs for cross side intervals